stdlib/public/runtime/Once.cpp

```diff
@@ -31,8 +31,10 @@ static_assert(std::is_same<swift_once_t, dispatch_once_t>::value,
 // The compiler generates the swift_once_t values as word-sized zero-initialized
 // variables, so we want to make sure swift_once_t isn't larger than the
 // platform word or the function below might overwrite something it shouldn't.
+#if !defined(__CYGWIN__)
```

Comment:
Well, then you need to keep the assertion and adjust `swift::swift_once` -- otherwise the initialization won't be correct.